src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -157,7 +136,7 @@ void AsyncLogWriter::write() {
     if (counter > 0) {
       stringStream ss;
       ss.print(UINT32_FORMAT_W(6) " messages dropped due to async logging", counter);
-      output->write_blocking(decorations, ss.as_string(false));
+      output->write_blocking(decorations, ss.freeze());
```

Comment:
Just to check but are we guaranteed that the ss content will be copied eventually?

Reply:
no, we don't copy the content of ss with ss.freeze() eventually. we just pass its underlying raw pointer to write_blocking. We don't need to copy here because the raw pointer is read-only. We destruct ss after write_blocking().